core/src/components/radio-group/test/compare-with/radio-group.e2e.ts

```diff
@@ -0,0 +1,91 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+/**
+ * This behavior does not vary across modes/directions.
+ */
+configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) => {
+  test.describe(title('radio-group: compare-with'), () => {
```

Comment:
Since we're testing the logic of radio group rather than the rendering, we can probably make these spec tests instead (they'll run a lot faster than the E2E tests too)

I migrated the first two E2E tests to give you an idea of how they should look:

```tsx
import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { RadioGroup } from '../radio-group';
import { Radio } from '../../radio/radio';

describe('radio-group: compareWith', () => {
  it('should correctly set value when using compareWith string', async () => {
    const page = await newSpecPage({
      components: [RadioGroup, Radio],
      template: () => (
        <ion-radio-group compare-with="value" value={{ label: 'Blue', value: 'blue' }}>
          <ion-radio value={{ label: 'Red', value: 'red' }}>Red</ion-radio>
          <ion-radio value={{ label: 'Blue', value: 'blue' }}>Blue</ion-radio>
          <ion-radio value={{ label: 'Green', value: 'green' }}>Green</ion-radio>
        </ion-radio-group>
      ),
    });

    const radioGroup = page.body.querySelector('ion-radio-group')!;

    expect(radioGroup.value).toEqual({
      label: 'Blue',
      value: 'blue',
    });
  });
  it('should correctly set value when using compareWith function', async () => {
    const page = await newSpecPage({
      components: [RadioGroup, Radio],
      template: () => (
        <ion-radio-group compare-with={(a, b) => a.value === b.value} value={{ label: 'Blue', value: 'blue' }}>
          <ion-radio value={{ label: 'Red', value: 'red' }}>Red</ion-radio>
          <ion-radio value={{ label: 'Blue', value: 'blue' }}>Blue</ion-radio>
          <ion-radio value={{ label: 'Green', value: 'green' }}>Green</ion-radio>
        </ion-radio-group>
      ),
    });

    const radioGroup = page.body.querySelector('ion-radio-group')!;

    expect(radioGroup.value).toEqual({
      label: 'Blue',
      value: 'blue',
    });
  });
});
```

(Note I'm using TSX here, so you'll want to make sure the test file has a `.tsx` extension not `.ts`. See https://stenciljs.com/docs/unit-testing#newspecpage for more info.

Reply:
Let's also add a test for when `compareWith` is `null`

Reply:
I'm instead changing these to all have a click that changes the value. Rewriting the tests like your example made it clear that I was just testing that the value was what I had set it to... which isn't really what I intended to test.

```
    radioGroup.value = options[1];
    expect(radioGroup.value).toEqual(options[1]);
```

is basically the test I wrote 🤦🏻

Reply:
Glad to hear that we are able to improve the test coverage, though my main point was that writing E2E tests are a bit overkill for what we need because aren't benefitting from cross-browser testing. They are fine as is, but they're going to take significantly longer than spec tests.

For example, running only the first E2E test takes 4.5s on my computer whereas writing it as a spec test takes 0.43s:

```js
import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { RadioGroup } from '../../radio-group';
import { Radio } from '../../../radio/radio';

describe('radio-group: compareWith', () => {
  it('should correctly set value when using compareWith string', async () => {
    const page = await newSpecPage({
      components: [RadioGroup, Radio],
      template: () => (
        <ion-radio-group compare-with="value">
          <ion-radio value={{ label: 'Red', value: 'red' }}>Red</ion-radio>
          <ion-radio value={{ label: 'Blue', value: 'blue' }}>Blue</ion-radio>
          <ion-radio value={{ label: 'Green', value: 'green' }}>Green</ion-radio>
        </ion-radio-group>
      ),
    });

    const radioGroup = page.body.querySelector('ion-radio-group')!;
    const radios = page.body.querySelectorAll('ion-radio');

    await radios[2].click();
    await page.waitForChanges();

    expect(radioGroup.value).toEqual({
      label: 'Green',
      value: 'green',
    });
    expect(radios[2]).toHaveAttribute('aria-checked', 'true');
  });
});
```

Reply:
Oh, ok. None of the spec tests I looked at had any interactivity in them (e.g. clicking), so I didn't know that was the sort of thing we would want to move into those. I can rewrite them 👍🏻